### PR TITLE
Use the `UsageFlags` from TCGC directly in our emitter

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/4820
Fixes https://github.com/Azure/autorest.csharp/issues/4734

This PR changes our the converter logic in our emitter, so that we do not need to convert the usage flags from TCGC to ours, we directly use their result. As part of the motivation, a few workaround logic to "fix" the usages has been removed.

### CODE REVIEW

File: packages/http-client-csharp/emitter/src/lib/client-model-builder.ts [hunk 1]
```diff
@@ -245,20 +215,20 @@ export function createModelForService(
...
+      const container = getHttpOperationWithCache(sdkContext, operations[0]).container;
       clientDesc = getDoc(program, container) ?? "";
     }
 
-    const inputClient: InputClient = {
+    const inputClient = {
       Name: getClientName(client),
       Description: clientDesc,
-      Operations: [],
+      Operations: [] as InputOperation[],
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/emitter/src/lib/client-model-builder.ts:225]: we should revert this change.

File: packages/http-client-csharp/emitter/src/lib/model.ts [hunk 2]
```diff
@@ -99,197 +87,29 @@ export function getInputType(
...
-              affectedReturnTypes.add(element);
-            });
-          }
-        }
-      }
-    }
+  // TODO -- remove when https://github.com/Azure/typespec-azure/issues/1150 is resolved
+  // TCGC has a bug that sometimes the name of body type constructed back from spread parameters does not have a name, here we add a name for it.
+  if (operation && sdkType.kind === "model" && sdkType.name === "") {
+    sdkType.name = `${getOperationFullName(operation)}Request`;
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/emitter/src/lib/model.ts:94]: this should have been fixed in the latest version of typespec/TCGC.
We should remove this once we adopted the new version of typespec.

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/EnumProviderTests.cs [hunk 3]
```diff
@@ -63,7 +63,7 @@ public void BuildEnumType_ValidateFloatBasedFixedEnum()
         {
             MockHelpers.LoadMockPlugin(createCSharpTypeCore: (inputType) => typeof(float));
 
-            var input = new InputEnumType("mockInputEnum", "mockNamespace", "public", null, "The mock enum", InputModelTypeUsage.RoundTrip, new InputPrimitiveType(InputPrimitiveTypeKind.Float32), [new InputEnumTypeValue("One", 1f, null), new InputEnumTypeValue("Two", 2f, null)], false);
+            var input = new InputEnumType("mockInputEnum", "mockNamespace", "public", null, "The mock enum", InputModelTypeUsage.Input | InputModelTypeUsage.Output, new InputPrimitiveType(InputPrimitiveTypeKind.Float32), [new InputEnumTypeValue("One", 1f, null), new InputEnumTypeValue("Two", 2f, null)], false);
```
Comment (archerzz) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/EnumProviderTests.cs:66]: Is that possible we provide a shortcut to `RoundTrip`?

**Review by archerzz — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/EnumProviderTests.cs (hunk 3 above)
Comment (archerzz) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/EnumProviderTests.cs:66]: oops, already discussed in another pr

File: packages/http-client-csharp/emitter/src/lib/converter.ts [hunk 4]
```diff
@@ -312,7 +311,7 @@ function fromSdkConstantType(
       Deprecated: undefined,
       Description: `The ${enumName}`, // TODO -- what should we put here?
       IsExtensible: true,
-      Usage: "None", // will be updated later
+      Usage: literalTypeContext.Usage ?? UsageFlags.Input,
```
Comment (chunyu3) [inline packages/http-client-csharp/emitter/src/lib/converter.ts:314]: Why do we set `Input` by default?

File: packages/http-client-csharp/emitter/test/Unit/utils/test-util.ts [hunk 5]
```diff
@@ -132,8 +128,5 @@ export function createNetSdkContext(
   program: EmitContext<NetEmitterOptions>
 ): SdkContext<NetEmitterOptions> {
   Logger.initialize(program.program, LoggerLevel.INFO);
-  const sdkContext = createSdkContext(program, "@azure-tools/typespec-azure");
-  // initialize TCGC
-  getAllModels(sdkContext);
-  return sdkContext;
+  return createSdkContext(program, "@azure-tools/typespec-azure");
```
Comment (chunyu3) [inline packages/http-client-csharp/emitter/test/Unit/utils/test-util.ts:131]:
```suggestion
  return createSdkContext(program, "@typespec/http-client-csharp");
```

**Review by chunyu3 — APPROVED** [review]
